src/OpenTelemetry.Contrib.Instrumentation.Quartz/Implementation/QuartzDiagnosticListener.cs

```diff
@@ -0,0 +1,128 @@
+using System.Linq;
+using System.Reflection;
+using OpenTelemetry.Instrumentation;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Contrib.Instrumentation.Quartz.Implementation
+{
+    internal sealed class QuartzDiagnosticListener : ListenerHandler
+    {
+        internal static readonly AssemblyName AssemblyName = typeof(QuartzDiagnosticListener).Assembly.GetName();
+        internal static readonly string ActivitySourceName = AssemblyName.Name;
+        internal static readonly Version Version = AssemblyName.Version;
+        internal static readonly ActivitySource ActivitySource = new ActivitySource(ActivitySourceName, Version.ToString());
+
+        private readonly QuartzInstrumentationOptions options;
+
+        public QuartzDiagnosticListener(string sourceName, QuartzInstrumentationOptions options)
+            : base(sourceName)
+        {
+            this.options = options ?? throw new ArgumentNullException(nameof(options));
+        }
+
+        public override void OnStartActivity(Activity activity, object payload)
+        {
+            // By this time, samplers have already run and
+            // activity.IsAllDataRequested populated accordingly.
+
+            if (Sdk.SuppressInstrumentation)
+            {
+                return;
+            }
+
+            if (activity.IsAllDataRequested)
+            {
+                if (this.options.TracedOperations != null && !this.options.TracedOperations.Contains(activity.OperationName))
+                {
+                    QuartzInstrumentationEventSource.Log.RequestIsFilteredOut(activity.OperationName);
+                    activity.IsAllDataRequested = false;
+                    return;
+                }
+
+                activity.DisplayName = this.GetDisplayName(activity);
+
+                ActivityInstrumentationHelper.SetActivitySourceProperty(activity, ActivitySource);
+                ActivityInstrumentationHelper.SetKindProperty(activity, this.GetActivityKind(activity));
+            }
+        }
+
+        public override void OnStopActivity(Activity activity, object payload)
+        {
+            if (activity.IsAllDataRequested)
+            {
+                try
+                {
+                    return;
+                }
+                catch (Exception ex)
+                {
+                    QuartzInstrumentationEventSource.Log.EnrichmentException(ex);
+                }
+            }
+        }
+
+        public override void OnException(Activity activity, object payload)
+        {
+            if (!this.options.TracedOperations.Contains(activity.OperationName))
+            {
+                return;
+            }
+
+            if (!(payload is Exception exception))
+            {
+                QuartzInstrumentationEventSource.Log.NullPayload(nameof(QuartzDiagnosticListener), nameof(this.OnStopActivity));
+                return;
+            }
+
+            activity.AddTag(SemanticConventions.AttributeExceptionType, "true");
```

Comment:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Api/Trace/ActivityExtensions.cs#L77 For exception reporting, please use this method.

Reply:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/exceptions.md#attributes

Reply:
updated to use activity extension